open-api/rest-catalog-open-api.yaml

```diff
@@ -1559,6 +1578,22 @@ components:
             type: string
           description:
             Properties that should be used as default configuration; applied before client configuration.
+        capabilities:
+          type: array
+          uniqueItems: true
+          items:
+            type: string
+            enum:
```

Comment:
How is this extensible and backwards/forwards compatible?
I fear that using a fixed enum will break older clients talking to servers that expose capabilities that these older clients do not know about.

Reply:
In the spec this is just an enumeration of options. The actual impl will make sure that this is forward/backward compatible. 

That means old clients talking to a newer server (that returns `capabilities`) wouldn't read `capabilities`. Newer clients (that read `capabilities`) talking to old servers (not returning `capabilities`) would not call any of the optional routes and/or use those capabilities.

Additionally, the implementation would make sure that adding new capabilities is safe and also forward/backward compatible. If a server sends back `[A, B, C, D]` but the client only knows about `[A, B]`, then `[C, D]` will be treated as `UNKNOWN` capabilities. 
If a client knows about `[A, B, C, D]` but the server only sends back `[A, B]`, the client knows that `[C, D]` are not available.

All of this will be handled at the JSON level when parsing `capabilities`.

Reply:
> In the spec this is just an enumeration of options. The actual impl will make sure that this is forward/backward compatible.
> All of this will be handled at the JSON level when parsing capabilities in a ConfigResponse.

You're focused on Java with a lot of custom and manual parsing code here.

This enum becomes a _language native_ enum. So all other languages need to implement custom logic. Even partial code generation becomes impossible - or is at high risk of being broken.

Adding another value would violate the (older) open-api spec of older clients/servers.

Reply:
I'm still not convinced that using `enum` here, especially for capabilities that change over time, is a good choice. Validations will legitly fail - and partly generated code as well - breaking back/forward  compatibility.

Reply:
I agree with @snazy here. Even if we don't recommend it, generated code could attempt to validate the enum and break if a service implements a newer version than a client. We should make this a list of strings and document that it corresponds to tags in the spec in a description.

Reply:
I've updated this to be plain strings and added a short description